api/desecapi/renderers.py

```diff
@@ -11,11 +12,23 @@ def render(self, data, media_type=None, renderer_context=None):
         response = renderer_context.get('response')
 
         if response and response.exception:
-            # TODO enable the renderer to handle responses where reponse.data is a list
-            if not isinstance(data, dict) or data.get('detail', None) is None:
-                raise ValueError('Expected response.data to be a dict with error details in response.data[\'detail\'], '
-                                 'but got %s:\n\n%s' % (type(response.data), response.data))
             response['Content-Type'] = 'text/plain'
-            return data['detail']
+
+            try:
+                return data['detail']
+            except KeyError:
```

Comment:
If `data` is a list of dicts, this raises `TypeError` which is not caught here. This would immediately raise and not run the following code. That is probably not intended. Is my understanding correct?

Reply:
Yes, fixed